aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -616,6 +616,395 @@ Tensor matrix_rank(const Tensor& self, bool symmetric) {
+  auto U = at::empty({0}, input.options());
+  const auto real_dtype = at::toValueType(input.scalar_type());
+  auto S = at::empty({0}, input.options().dtype(real_dtype));
+  auto Vh = at::empty({0}, input.options());
+  auto rank = at::empty({0}, input.options().dtype(ScalarType::Long));
+  linalg_svd_rank_revealing_out(input, atol, rtol, full_matrices, U, S, Vh, rank);
+  return std::make_tuple(std::move(U), std::move(S), std::move(Vh), std::move(rank));
+}
+
+std::tuple<Tensor, Tensor, Tensor, Tensor>
+linalg_svd_rank_revealing(
+    const Tensor& input,
+    c10::optional<double> atol,
+    c10::optional<double> rtol,
+    bool full_matrices) {
+  Tensor atol_tensor, rtol_tensor;
+  std::tie(atol_tensor, rtol_tensor) = get_atol_rtol(input, atol, rtol);
+  return at::linalg_svd_rank_revealing(input, atol_tensor, rtol_tensor, full_matrices);
+}
+
+std::tuple<Tensor, Tensor, Tensor, Tensor>
+linalg_svd_rank_revealing(
+    const Tensor& input,
+    const Tensor& tol,
+    bool full_matrices) {
+  const auto rtol = at::zeros({}, tol.options());
+  return at::linalg_svd_rank_revealing(input, tol, rtol, full_matrices);
+}
+
+std::tuple<Tensor, Tensor, Tensor, Tensor>
+linalg_svd_rank_revealing(
+    const Tensor& input,
+    double tol,
+    bool full_matrices) {
+  return at::linalg_svd_rank_revealing(input, tol, /*rtol=*/0.0, full_matrices);
+}
+
+std::tuple<Tensor, Tensor, Tensor, Tensor>
+linalg_svd_rank_restricted(
+    const Tensor& input,
+    const c10::optional<Tensor>& atol,
+    const c10::optional<Tensor>& rtol,
+    bool full_matrices) {
+  Tensor U, S, Vh, rank, unique_rank;
+  std::tie(U, S, Vh, rank, unique_rank) = at::_linalg_svd_rank_restricted_helper(
+      input, atol, rtol, full_matrices,
+      /*compute_unique_rank=*/input.numel() ? true : false
+  );
+
+  // fast path for emtpy inputs
+  if (!input.numel()) {
+    return std::make_tuple(std::move(U), std::move(S), std::move(Vh), std::move(rank));
+  }
+
+  auto U_rank_restricted = at::zeros_like(U);
+  auto S_rank_restricted = at::zeros_like(S);
+  auto Vh_rank_restricted = at::zeros_like(Vh);
+
+  const auto n_batch_dims = input.dim() - 2;
+  const auto m = input.size(-2);
+  const auto n = input.size(-1);
+  const auto k = std::max(m, n);
+  const auto U_dim = U.size(-1);
+  const auto S_len = S.size(-1);
+  const auto V_dim = Vh.size(-2);
+
+  using at::indexing::Slice;
+  using at::indexing::TensorIndex;
+
+  for (const auto i : c10::irange(unique_rank.numel())) {
+    const auto r = unique_rank.select(0, i).item<int64_t>();
+    // init a mask by setting elements r:k to zero
+    // NOTE: for performance reasons it would make sense to store these masks
+    // in a single tensor. However, a view + zero_ on a buffer breaks CompositeCompliance tests.
+    // TODO: fix that once and if things with the autograd composite compliance become
+    // more flexible.
+    auto r_mask = at::ones({k}, input.options().dtype(at::kBool));
+    r_mask.narrow(-1, r, k - r).zero_();
```

Comment:
Not valid anymore. The tests fail anyway after merging master.

Reply:
What is the problem? That the view + in-place breaks composite compliance?

Reply:
Yes.

Reply:
So, from looking at the code, this looks like it shouldn't cause the composite compliance check to fail. I'm getting "RuntimeError: This operator is not CompositeImplicitAutograd compliant: the storage_offset of the tensor was modified directly without going through the PyTorch dispatcher." when I run this locally, are you seeing the same error?

Reply:
Thanks for looking into it! Yes, I am seeing the same issue, but only after pulling master. That was not causing issues like a week ago.

Reply:
Not relevant atm, now the tests pass again.

Reply:
Had to replace in-place `zero_` with out-of-place equivalents and now all works.